OpenRA.Mods.Common/Traits/Attack/AttackFollow.cs

```diff
@@ -93,7 +94,8 @@ public override Activity Tick(Actor self)
 				if (self.IsDisabled())
 					return this;
 
-				var weapon = attack.ChooseArmamentsForTarget(target, forceAttack).FirstOrDefault();
+				var attack = SelectActiveAttack(target, forceAttack).FirstOrDefault();
+				var weapon = attack.Value;
```

Comment:
Turns out KeyValuePair can never be null so it will return a default value, if the default values for the key and value is null then you can ignore this comment.

Though you could check if IEnumerable.Any() returns True before doing First but I guess thats unnecessary.